docker/owlbot/java/cloudbuild.yaml

```diff
@@ -12,14 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 steps:
-  - name: 'gcr.io/cloud-builders/docker'
-    args: [ 'build',
-      '-t', 'gcr.io/$PROJECT_ID/owlbot-java:$SHORT_SHA',
-      '-t', 'gcr.io/$PROJECT_ID/owlbot-java:latest',
-      '-f', 'docker/owlbot/java/Dockerfile', '.' ]
+  - name: "gcr.io/cloud-builders/docker"
+    args: [ "build",
+      "-t", "gcr.io/$PROJECT_ID/owlbot-java:$SHORT_SHA",
+      "-t", "gcr.io/$PROJECT_ID/owlbot-java:latest",
+      "-f", "docker/owlbot/java/Dockerfile", "." ]
+    id: "build"
   - name: gcr.io/gcp-runtimes/container-structure-test
     args:
       ["test", "--image", "gcr.io/$PROJECT_ID/owlbot-java:$SHORT_SHA", "--config", "docker/owlbot/java/container_test.yaml"]
+    waitFor: ["build"]
+
+  # end-to-end tests
+  - name: "gcr.io/$PROJECT_ID/owlbot-java:$SHORT_SHA"
+    dir: docker/owlbot/java/tests/new-client/input
+    id: "new-client-run"
+    waitFor: ["build"]      
+  - name: "bash"
+    args: ["diff", "-rw", "input", "golden"]
```

Comment:
Will this exit `1` if the output from `new-client/input` differs from `golden`?

Perhaps add a note in the README for the container, regarding how to create a new golden fixture?

Reply:
> Will this exit `1` if the output from `new-client/input` differs from `golden`?

It does (maybe not with `1`, but definitely a non-zero exit code) and will cause the build to fail.

> Perhaps add a note in the README for the container, regarding how to create a new golden fixture?

Will do.